okio-files/src/jvmMain/kotlin/okio/JvmSystemFilesystem.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import okio.Path.Companion.toPath
+
+object JvmSystemFilesystem : Filesystem() {
+  override val cwd: Path
+    get() {
+      val userDir = System.getProperty("user.dir") ?: error("user.dir system property missing?!")
```

Comment:
[Coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html#functions-vs-properties) state that:

- Properties shouldn't throw
- Properties should be expected to return the same result if the object state hasn't changed

Reply:
The throwing case is extremely unlikely;  it only happens if the System properties are edited to remove this built-in property. But I wanted it to crash with a nice message rather than an NullPointerException in that rare case.

Reply:
As for the changing value, again it's unlikely to change. But if it does it’s because something in the same process changed it. Think of this property as having a `private set` that manipulate it.